### PR TITLE
Recorder module

### PR DESCRIPTION
**Pull request by korshaknn**
Added 2 new modules:
- common. Has BaseActivity, FpsPerformanceView and ViewUtils. Common gradle dependencies moved here
- recorder. A separate app with recording logic. Depend on common module

App module (Teaser) also depends on common one.

Linked to [1352](https://app.zenhub.com/workspaces/vision-sdk-5ca52e6c027e3e7234ac7512/issues/mapbox/mapbox-vision/1352)

### CODE REVIEW

**Comment by korshaknn** [comment]
@yunikkk @RingerJK @alexander-kulikovskii @kiryldz ptal

File: app/src/main/java/com/mapbox/vision/examples/activity/main/MainActivity.kt [hunk 1]
```diff
@@ -267,34 +263,16 @@ class MainActivity : AppCompatActivity() {
...
-            requestPermissions(getRequiredPermissions(), PERMISSIONS_REQUEST_CODE)
-        } else {
-            onPermissionsGranted()
-        }
+    override fun initViews() {
+        setContentView(R.layout.activity_main)
     }
 
-    private fun onPermissionsGranted() {
+    public override fun onPermissionsGranted() {
```
Comment (kiryldz) [inline app/src/main/java/com/mapbox/vision/examples/activity/main/MainActivity.kt]: Is `public` needed here?

File: recorder/src/main/java/com/mapbox/vision/recorder/RecordingActivity.kt [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+                fps_performance_view.showInfo(frameStatistics, appModelPerformanceConfig)
+            }
+        }
+    }
+
+    override fun initViews() {
+        setContentView(R.layout.activity_recording)
+    }
+
+    public override fun onPermissionsGranted() {
```
Comment (kiryldz) [inline recorder/src/main/java/com/mapbox/vision/recorder/RecordingActivity.kt]: Is `public` needed here?

File: recorder/src/main/java/com/mapbox/vision/recorder/RecordingView.kt [hunk 3]
```diff
@@ -0,0 +1,44 @@
+package com.mapbox.vision.recorder
+
+import android.content.Context
+import android.util.AttributeSet
+import android.view.LayoutInflater
+import android.widget.LinearLayout
+import androidx.annotation.DrawableRes
+import androidx.annotation.StringRes
+import kotlinx.android.synthetic.main.view_recording.view.*
```
Comment (kiryldz) [inline recorder/src/main/java/com/mapbox/vision/recorder/RecordingView.kt]: ktlint?

File: recorder/src/main/java/com/mapbox/vision/recorder/RecordingView.kt [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+class RecordingView @JvmOverloads constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    defStyle: Int = 0,
+    defStyleRes: Int = 0
+) : LinearLayout(context, attrs, defStyle, defStyleRes) {
+
+    companion object {
```
Comment (kiryldz) [inline recorder/src/main/java/com/mapbox/vision/recorder/RecordingView.kt]: Guess we can remove it

**Comment by korshaknn** [comment]
@yunikkk @RingerJK  should we use Crashlytics in Recorder?

File: recorder/src/main/java/com/mapbox/vision/recorder/RecordingActivity.kt [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+import com.mapbox.vision.common.view.show
+import com.mapbox.vision.mobile.core.interfaces.VisionEventsListener
+import com.mapbox.vision.mobile.core.models.Camera
+import com.mapbox.vision.mobile.core.models.Country
+import com.mapbox.vision.mobile.core.utils.SystemInfoUtils
+import com.mapbox.vision.mobile.core.utils.snapdragon.SupportedSnapdragonBoards
+import com.mapbox.vision.performance.ModelPerformance
+import com.mapbox.vision.performance.ModelPerformanceConfig
+import com.mapbox.vision.utils.VisionLogger
+import kotlinx.android.synthetic.main.activity_recording.*
```
Comment (kiryldz) [inline recorder/src/main/java/com/mapbox/vision/recorder/RecordingActivity.kt]: ktlint

File: app/src/main/java/com/mapbox/vision/examples/activity/main/MainActivity.kt (hunk 1 above)
Comment (korshaknn) [reply app/src/main/java/com/mapbox/vision/examples/activity/main/MainActivity.kt]: yep, a base class is stored in a separate module

File: recorder/src/main/java/com/mapbox/vision/recorder/RecordingActivity.kt (hunk 2 above)
Comment (korshaknn) [reply recorder/src/main/java/com/mapbox/vision/recorder/RecordingActivity.kt]: yep, a base class is stored in a separate module

File: recorder/src/main/java/com/mapbox/vision/recorder/RecordingView.kt (hunk 4 above)
Comment (korshaknn) [reply recorder/src/main/java/com/mapbox/vision/recorder/RecordingView.kt]: done

File: recorder/src/main/java/com/mapbox/vision/recorder/RecordingActivity.kt (hunk 5 above)
Comment (korshaknn) [reply recorder/src/main/java/com/mapbox/vision/recorder/RecordingActivity.kt]: done

File: recorder/src/main/java/com/mapbox/vision/recorder/RecordingView.kt (hunk 3 above)
Comment (korshaknn) [reply recorder/src/main/java/com/mapbox/vision/recorder/RecordingView.kt]: done

File: common/src/main/AndroidManifest.xml [hunk 6]
```diff
@@ -0,0 +1,2 @@
+<manifest xmlns:android="http://schemas.android.com/apk/res/android"
+    package="com.mabpox.vision.common" />
```
Comment (yunikkk) [inline common/src/main/AndroidManifest.xml]: Suggest `com.mabpox.vision.teaser.common` instead?

File: common/src/main/java/com/mapbox/vision/common/BaseActivity.kt [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+import android.text.Html
+import android.view.WindowManager
+import android.widget.Toast
+import androidx.appcompat.app.AppCompatActivity
+import androidx.core.content.ContextCompat
+import com.mapbox.vision.mobile.core.utils.SystemInfoUtils
+import com.mapbox.vision.mobile.core.utils.snapdragon.SupportedSnapdragonBoards
+import com.mapbox.vision.utils.VisionLogger
+
+abstract class BaseActivity : AppCompatActivity() {
```
Comment (yunikkk) [inline common/src/main/java/com/mapbox/vision/common/BaseActivity.kt]: BaseVisionActivity to easy understanding for external devs to understand it's our custom base activity?

File: recorder/src/main/java/com/mapbox/vision/recorder/RecordingActivity.kt [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+import com.mapbox.vision.performance.ModelPerformanceConfig
+import java.text.SimpleDateFormat
+import java.util.Date
+import java.util.Locale
+import kotlinx.android.synthetic.main.activity_recording.*
+
+class RecordingActivity : BaseActivity() {
+
+    companion object {
+        private val BASE_SESSION_PATH = "${Environment.getExternalStorageDirectory().absolutePath}/Telemetry"
```
Comment (yunikkk) [inline recorder/src/main/java/com/mapbox/vision/recorder/RecordingActivity.kt]: Now that I think about name it doesn't seem like an understandable one. Seems smth with `Vision` will work much better, `VisionTelemetry`/`MapboxVisionTelemetry` etc. ?

File: recorder/src/main/java/com/mapbox/vision/recorder/RecordingActivity.kt [hunk 9]
```diff
@@ -0,0 +1,119 @@
...
+    private var visionManagerWasInit = false
+
+    private var appModelPerformanceConfig: ModelPerformanceConfig = ModelPerformanceConfig.Merged(
+        performance = ModelPerformance.Off
+    )
+
+    private val visionEventsListener = object : VisionEventsListener {
+
+        override fun onCountryUpdated(country: Country) {
+            println("Country $country")
```
Comment (yunikkk) [inline recorder/src/main/java/com/mapbox/vision/recorder/RecordingActivity.kt]: Remove?

**Comment by yunikkk** [comment]
Also think README should be updated and docs, too

File: common/src/main/java/com/mapbox/vision/common/BaseActivity.kt (hunk 7 above)
Comment (korshaknn) [reply common/src/main/java/com/mapbox/vision/common/BaseActivity.kt]: done

File: recorder/src/main/java/com/mapbox/vision/recorder/RecordingActivity.kt (hunk 8 above)
Comment (korshaknn) [reply recorder/src/main/java/com/mapbox/vision/recorder/RecordingActivity.kt]: done

File: recorder/src/main/java/com/mapbox/vision/recorder/RecordingActivity.kt (hunk 9 above)
Comment (korshaknn) [reply recorder/src/main/java/com/mapbox/vision/recorder/RecordingActivity.kt]: done

**Pull request merged**